traffic_ops/traffic_ops_golang/riak.go

```diff
@@ -22,17 +41,169 @@ type URISignerKeyset struct {
 	Keys       []jwk.EssentialHeader `json:"keys"`
 }
 
-func getStringValueFromRiakObject(resp *riak.FetchValueResponse) (string, error) {
-	var obj *riak.Object
+// deletes an object from riak storage
+func deleteObject(key string, bucket string, cluster *riak.Cluster) error {
+	// build store command and execute.
+	cmd, err := riak.NewDeleteValueCommandBuilder().
+		WithBucket(bucket).
+		WithKey(key).
+		Build()
+	if err != nil {
+		return err
+	}
+	if err := cluster.Execute(cmd); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// fetch an object from riak storage
+func fetchObjectValues(key string, bucket string, cluster *riak.Cluster) ([]*riak.Object, error) {
+	// build the fetch command
+	cmd, err := riak.NewFetchValueCommandBuilder().
+		WithBucket(bucket).
+		WithKey(key).
+		Build()
+	if err != nil {
+		return nil, err
+	}
+
+	if err = cluster.Execute(cmd); err != nil {
+		return nil, err
+	}
+	fvc := cmd.(*riak.FetchValueCommand)
+
+	// no object found with given key
+	if fvc.Response == nil || fvc.Response.IsNotFound {
+		return nil, nil
+	}
+	return fvc.Response.Values, nil
+}
+
+// saves an object to riak storage
+func saveObject(obj *riak.Object, bucket string, cluster *riak.Cluster) error {
+	// build store command and execute.
+	cmd, err := riak.NewStoreValueCommandBuilder().
+		WithBucket(bucket).
+		WithContent(obj).
+		Build()
+	if err != nil {
+		return err
+	}
+	if err := cluster.Execute(cmd); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// returns a riak cluster of online riak nodes.
+func getRiakCluster(db *sqlx.DB, cfg Config) (*riak.Cluster, error) {
+	riakServerQuery := `
+		SELECT s.host_name, s.domain_name FROM server s 
+		INNER JOIN type t on s.type = t.id 
+		INNER JOIN status st on s.status = st.id 
+		WHERE t.name = 'RIAK' AND st.name = 'ONLINE'
+		`
+
+	if cfg.RiakAuthOptions == nil {
+		return nil, errors.New("ERROR: no riak auth information from riak.conf, cannot authenticate to any riak servers.")
+	}
+
+	var nodes []*riak.Node
+	rows, err := db.Query(riakServerQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	for rows.Next() {
+		var s tc.Server
+		var n *riak.Node
+		if err := rows.Scan(&s.HostName, &s.DomainName); err != nil {
+			return nil, err
+		}
+		addr := fmt.Sprintf("%s.%s:%d", s.HostName, s.DomainName, RiakPort)
+		nodeOpts := &riak.NodeOptions{
+			RemoteAddress: addr,
+			AuthOptions:   cfg.RiakAuthOptions,
+		}
+		nodeOpts.AuthOptions.TlsConfig.ServerName = fmt.Sprintf("%s.%s", s.HostName, s.DomainName)
+		n, err := riak.NewNode(nodeOpts)
+		if err != nil {
+			return nil, err
+		}
+		nodes = append(nodes, n)
+	}
+
+	if len(nodes) == 0 {
+		return nil, errors.New("ERROR: no available riak servers.")
+	}
+
+	opts := &riak.ClusterOptions{
+		Nodes: nodes,
+	}
+	cluster, err := riak.NewCluster(opts)
+
+	return cluster, err
+}
+
+// endpoint handler for fetching uri signing keys from riak
+func getUrisignkeysHandler(db *sqlx.DB, cfg Config) http.HandlerFunc {
```

Comment:
getUrisignkeysHandler should be getURIsignkeysHandler, URI is an acronym and should be all caps